_posts/2015-08-04-creating-systems-with-pipelining.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "Creating systems with pipelining"
+date: "2015-08-04"
+icon: wrench
+tags: ["pipelines", "virtual machines", "development systems"]
+categories: tech-stories
+authors: ["Carsten", "Karsten"]
+---
+
+If we talk about [pipelining](https://en.wikipedia.org/wiki/Pipeline_(software)) at ePages, we understand it as an automatic workflow.
+We are integrating many steps using different tools to reach a specific goal: creating a simple way to provide a highly specialised system.
+
+This system can be used for development purposes, external or even live deployment.
+The setup used here is a combination of different software and tools.
+One of them is [vmware](http://www.vmware.com/) with [vCenter](https://www.vmware.com/products/vcenter-server) and some [ESXi](https://www.vmware.com/products/vsphere/features/esxi-hypervisor) servers to offer and manage systems in a virtual environment.
+Another tool we use is called [i-doit](http://www.i-doit.org/) and is used for creation and documentation of inventories and network related information.
+As you may read in a previous [post](https://developer.epages.com/blog/2015/06/25/infrastructure-as-code.html) we often use [Jenkins](https://jenkins-ci.org/) for [Continuous Integration](http://www.martinfowler.com/articles/continuousIntegration.html), so it's one of our favourite tools to automate processes.
+The key of an automated OS install is the combination of [iPXE](http://ipxe.org/) and kickstart isos, for example preseed values for [Debian](https://wiki.debian.org/DebianInstaller/Preseed) or [Centos/RHEL](http://www.centos.org/docs/4/html/rhel-sag-en-4/s1-kickstart2-file.html).
+Last but not least when it comes to system specialisation [Linux](https://www.linux.com/) itself should be mentioned.
+To speak with the different interfaces, programming languages like [`Perl`](https://www.perl.org/) and [`Python`](https://www.python.org/) or scripting languages like [`PHP`](https://secure.php.net/) and [`bash`](http://www.gnu.org/software/bash/) are used.
+
+{% image blog-pipeline-tools.png %} {% endimage %}
+
+The basic idea is simple. First thing we need is a system.
+
+# The system
+
+This is fairly simple, due to the [vSphere Cloud Plugin](https://wiki.jenkins-ci.org/display/JENKINS/vSphere+Cloud+Plugin) for Jenkins, it is easy to construct a job building a basic virtual machine from a template.
```

Comment:
hm i could add a sentence before:
A virtualized hardware environment is the base for the system and managed by our vmware ESXi/vCenter Servers.
what do you think ?